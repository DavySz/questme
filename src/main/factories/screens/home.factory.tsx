import { HomeContainer } from "@presentation/screens";
import { ReactNode } from "react";

export function MakeHomeScreen(): ReactNode {
  return <HomeContainer />;
}
