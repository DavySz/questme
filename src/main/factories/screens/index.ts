export * from "./onboarding-factory";
export * from "./splash-factory";
export * from "./initial-factory";
export * from "./login-factory";
export * from "./sign-up-factory";
