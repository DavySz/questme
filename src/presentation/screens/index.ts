export * from "./home/home.container";
export * from "./splash/splash.container";
export * from "./onboarding/onboarding.container";
export * from "./initial/initial.container";
export * from "./login/login.container";
export * from "./sign-up/sign-up.container";
export * from "./reset-password/reset-password.container";
export * from "./new-password/new-password.container";
export * from "./sign-up-email";
