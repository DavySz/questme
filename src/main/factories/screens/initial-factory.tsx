import { InitialContainer } from "@presentation/screens";
import { ReactNode } from "react";

export function MakeInitialScreen(): ReactNode {
  return <InitialContainer />;
}
