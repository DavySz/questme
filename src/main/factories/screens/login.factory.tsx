import { LoginContainer } from "@presentation/screens";
import { ReactNode } from "react";

export function MakeLoginScreen(): ReactNode {
  return <LoginContainer />;
}
