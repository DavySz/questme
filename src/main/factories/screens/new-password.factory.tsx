import { NewPasswordContainer } from "@presentation/screens";
import { ReactNode } from "react";

export function MakeNewPasswordScreen(): ReactNode {
  return <NewPasswordContainer />;
}
