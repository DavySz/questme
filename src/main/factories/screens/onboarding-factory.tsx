import { OnboardingContainer } from "@presentation/screens";
import { ReactNode } from "react";

export function MakeOnboardingScreen(): ReactNode {
  return <OnboardingContainer />;
}
