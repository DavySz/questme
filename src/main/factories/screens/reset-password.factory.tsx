import { ResetPasswordContainer } from "@presentation/screens";
import { ReactNode } from "react";

export function MakeResetPasswordScreen(): ReactNode {
  return <ResetPasswordContainer />;
}
