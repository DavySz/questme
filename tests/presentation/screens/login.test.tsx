import { LoginContainer } from "@presentation/screens";
import { customRender, TestingPublicNavigator } from "../../utils";
import { fireEvent, render } from "@testing-library/react-native";
describe("Login", () => {
  test("Should render without errors", () => {
    const tree = customRender(<LoginContainer />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("Should go back to the previous screen when go back button is pressed", async () => {
    const { getByTestId, findByTestId } = render(
      <TestingPublicNavigator currentRoute="initial" />
    );

    const goToLoginButton = getByTestId("button-login");
    fireEvent.press(goToLoginButton);

    const goBackButton = getByTestId("go-back-button");
    fireEvent.press(goBackButton);

    const loginScreen = await findByTestId("initial-screen-content");

    expect(loginScreen).toBeTruthy();
  });
});
