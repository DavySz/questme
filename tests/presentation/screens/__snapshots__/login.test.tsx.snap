// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Should render without errors 1`] = `
<View
  onLayout={[Function]}
  style={
    {
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
    }
  }
  testID="key-board-safe-area"
>
  <View
    style={
      {
        "backgroundColor": "#EFEEFC",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      }
    }
    testID="default-container"
    variant="secondary"
  >
    <View
      style={
        {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "justifyContent": "center",
          "paddingBottom": 16,
          "paddingLeft": 16,
          "paddingRight": 16,
          "paddingTop": 16,
          "width": "100%",
        }
      }
    >
      <View
        style={
          {
            "left": 16,
            "position": "absolute",
            "top": 8,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "transparent",
              "borderBottomLeftRadius": 20,
              "borderBottomRightRadius": 20,
              "borderTopLeftRadius": 20,
              "borderTopRightRadius": 20,
              "flexDirection": "row",
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 16,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 16,
            }
          }
          testID="go-back-button"
        >
          <View
            side="none"
            style={{}}
            testID="icon-wrapper"
          >
            <SvgMock
              color="#0C092A"
              testID="icon-with-color"
            />
          </View>
        </View>
      </View>
      <Text
        color="neutral-black"
        style={
          {
            "color": "#0C092A",
            "fontFamily": "Rubik_500Medium",
            "fontSize": 24,
            "lineHeight": 36,
          }
        }
        testID="header-title"
        variant="heading-3"
      >
        Login
      </Text>
    </View>
    <RCTScrollView
      contentContainerStyle={
        {
          "flex": 1,
          "padding": 24,
        }
      }
      showsVerticalScrollIndicator={false}
      style={{}}
      testID="login-screen-content"
      variant="secondary"
    >
      <View>
        <View
          style={
            {
              "alignItems": "center",
              "gap": 16,
              "justifyContent": "center",
              "marginBottom": 24,
              "width": "100%",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FFF",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderColor": "#E6E6E6",
                "borderStyle": "solid",
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
                "width": "100%",
              }
            }
          >
            <View
              side="left"
              style={
                {
                  "marginRight": 16,
                }
              }
              testID="icon-wrapper"
            >
              <SvgMock
                testID="icon-without-color"
              />
            </View>
            <Text
              color="neutral-black"
              style={
                {
                  "color": "#0C092A",
                  "fontFamily": "Rubik_500Medium",
                  "fontSize": 16,
                  "lineHeight": 24,
                }
              }
              variant="body-normal-medium"
            >
              Login with Google
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#0056B2",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
                "width": "100%",
              }
            }
          >
            <View
              side="left"
              style={
                {
                  "marginRight": 16,
                }
              }
              testID="icon-wrapper"
            >
              <SvgMock
                testID="icon-without-color"
              />
            </View>
            <Text
              color="neutral-white"
              style={
                {
                  "color": "#FFF",
                  "fontFamily": "Rubik_500Medium",
                  "fontSize": 16,
                  "lineHeight": 24,
                }
              }
              variant="body-normal-medium"
            >
              Login with Facebook
            </Text>
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "center",
              "marginBottom": 24,
              "width": "100%",
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": "#E6E6E6",
                "flexDirection": "row",
                "height": 1,
                "marginBottom": 0,
                "marginLeft": 16,
                "marginRight": 16,
                "marginTop": 0,
                "width": "100%",
              }
            }
          />
          <Text
            color="neutral-gey-2"
            style={
              {
                "color": "#858494",
                "fontFamily": "Rubik_400Regular",
                "fontSize": 16,
                "lineHeight": 24,
              }
            }
            variant="body-normal-regular"
          >
            OR
          </Text>
          <View
            style={
              {
                "backgroundColor": "#E6E6E6",
                "flexDirection": "row",
                "height": 1,
                "marginBottom": 0,
                "marginLeft": 16,
                "marginRight": 16,
                "marginTop": 0,
                "width": "100%",
              }
            }
          />
        </View>
        <View
          style={
            {
              "gap": 16,
              "marginBottom": 24,
            }
          }
        >
          <View
            style={{}}
          >
            <View
              style={
                {
                  "marginBottom": 8,
                }
              }
            >
              <Text
                color="neutral-black"
                style={
                  {
                    "color": "#0C092A",
                    "fontFamily": "Rubik_400Regular",
                    "fontSize": 14,
                    "lineHeight": 20,
                  }
                }
                variant="body-small-regular"
              >
                Email Address
              </Text>
            </View>
            <View
              isFocused={false}
              style={
                {
                  "backgroundColor": "#FFF",
                  "borderBottomLeftRadius": 20,
                  "borderBottomRightRadius": 20,
                  "borderColor": "#EFEEFC",
                  "borderStyle": "solid",
                  "borderTopLeftRadius": 20,
                  "borderTopRightRadius": 20,
                  "borderWidth": 2,
                  "flexDirection": "row",
                  "gap": 16,
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                  "width": "100%",
                }
              }
              testID="input-wrapper"
            >
              <SvgMock
                color="#6A5AE0"
                testID="icon-with-color"
              />
              <TextInput
                inputMode="email"
                onBlur={[Function]}
                onFocus={[Function]}
                placeholder="Your email address"
                style={
                  {
                    "fontFamily": "Rubik_400Regular",
                    "fontSize": 16,
                    "width": "100%",
                  }
                }
                testID="custom-input"
              />
            </View>
          </View>
          <View
            style={{}}
          >
            <View
              style={
                {
                  "marginBottom": 8,
                }
              }
            >
              <Text
                color="neutral-black"
                style={
                  {
                    "color": "#0C092A",
                    "fontFamily": "Rubik_400Regular",
                    "fontSize": 14,
                    "lineHeight": 20,
                  }
                }
                variant="body-small-regular"
              >
                Password
              </Text>
            </View>
            <View
              isFocused={false}
              style={
                {
                  "backgroundColor": "#FFF",
                  "borderBottomLeftRadius": 20,
                  "borderBottomRightRadius": 20,
                  "borderColor": "#EFEEFC",
                  "borderStyle": "solid",
                  "borderTopLeftRadius": 20,
                  "borderTopRightRadius": 20,
                  "borderWidth": 2,
                  "flexDirection": "row",
                  "gap": 16,
                  "paddingBottom": 16,
                  "paddingLeft": 16,
                  "paddingRight": 16,
                  "paddingTop": 16,
                  "width": "100%",
                }
              }
              testID="input-wrapper"
            >
              <SvgMock
                testID="icon-without-color"
              />
              <TextInput
                onBlur={[Function]}
                onFocus={[Function]}
                placeholder="Your password"
                style={
                  {
                    "fontFamily": "Rubik_400Regular",
                    "fontSize": 16,
                    "width": "100%",
                  }
                }
                testID="custom-input"
              />
            </View>
          </View>
        </View>
        <View
          style={
            {
              "alignItems": "center",
              "gap": 16,
              "justifyContent": "center",
              "marginBottom": 24,
              "width": "100%",
            }
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#6A5AE0",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
                "width": "100%",
              }
            }
          >
            <Text
              color="neutral-white"
              style={
                {
                  "color": "#FFF",
                  "fontFamily": "Rubik_500Medium",
                  "fontSize": 16,
                  "lineHeight": 24,
                }
              }
              variant="body-normal-medium"
            >
              Login
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "transparent",
                "borderBottomLeftRadius": 20,
                "borderBottomRightRadius": 20,
                "borderTopLeftRadius": 20,
                "borderTopRightRadius": 20,
                "flexDirection": "row",
                "justifyContent": "center",
                "opacity": 1,
                "paddingBottom": 16,
                "paddingLeft": 16,
                "paddingRight": 16,
                "paddingTop": 16,
                "width": "100%",
              }
            }
            testID="button-reset-password"
          >
            <Text
              color="primary"
              style={
                {
                  "color": "#6A5AE0",
                  "fontFamily": "Rubik_500Medium",
                  "fontSize": 16,
                  "lineHeight": 24,
                }
              }
              variant="body-normal-medium"
            >
              Forgot password?
            </Text>
          </View>
        </View>
        <Text
          align="center"
          color="neutral-gey-2"
          style={
            {
              "color": "#858494",
              "fontFamily": "Rubik_400Regular",
              "fontSize": 14,
              "lineHeight": 20,
              "textAlign": "center",
            }
          }
          variant="body-small-regular"
        >
          By continuing, you agree to the Terms of
           
          <Text
            color="neutral-gey-1"
            style={
              {
                "color": "#49465F",
                "fontFamily": "Rubik_500Medium",
                "fontSize": 14,
                "lineHeight": 20,
              }
            }
            variant="body-small-medium"
          >
            Services & Privacy Policy.
          </Text>
        </Text>
      </View>
    </RCTScrollView>
  </View>
</View>
`;
